qa/tasks/cephfs/xfstests_dev.py

```diff
@@ -133,25 +135,54 @@ def create_reqd_users(self):
                                        check_status=False)
 
     def write_local_config(self):
-        from os.path import join
-        from textwrap import dedent
+        if isinstance(self.mount_a, KernelMount):
+            conf_contents = self._gen_conf_for_kernel_mnt()
+        elif isinstance(self.mount_a, (FuseMount, LocalFuseMount)):
+            conf_contents = self._gen_conf_for_fuse_mnt()
 
+        self.mount_a.client_remote.write_file(join(self.repo_path,
+                                                   'local.config'),
+                                              conf_contents, sudo=True)
+
+    # generate local.config for kernel cephfs mounts
+    def _gen_conf_for_kernel_mnt(self):
         mon_sock = self.fs.mon_manager.get_msgrv1_mon_socks()[0]
-        self.test_dev = mon_sock + ':/' + self.test_dirname
-        self.scratch_dev = mon_sock + ':/' + self.scratch_dirname
+        test_dev = mon_sock + ':/' + self.test_dirname
+        scratch_dev = mon_sock + ':/' + self.scratch_dirname
 
-        xfstests_config_contents = dedent('''\
+        return dedent(f'''\
             export FSTYP=ceph
-            export TEST_DEV={}
-            export TEST_DIR={}
-            #export SCRATCH_DEV={}
-            #export SCRATCH_MNT={}
-            export TEST_FS_MOUNT_OPTS="-o name=admin,secret={}"
-            ''').format(self.test_dev, self.test_dirs_mount_path, self.scratch_dev,
-                        self.scratch_dirs_mount_path, self.get_admin_key())
-
-        self.mount_a.client_remote.write_file(join(self.repo_path, 'local.config'),
-                                              xfstests_config_contents, sudo=True)
+            export TEST_DEV={test_dev}
+            export TEST_DIR={self.test_dirs_mount_path}
+            export TEST_FS_MOUNT_OPTS="-o name=admin,secret={self.get_admin_key()}"
+            #export SCRATCH_DEV={scratch_dev}
+            #export SCRATCH_MNT={self.scratch_dirs_mount_path}''')
+
+    # generate local.config for FUSE cephfs mounts
+    def _gen_conf_for_fuse_mnt(self):
+        mon_sock = self.fs.mon_manager.get_msgrv1_mon_socks()[0]
+        test_dev = 'ceph-fuse'
+        scratch_dev = ''
+        # XXX: Please note that ceph_fuse_bin_path is not ideally required
+        # because ceph-fuse binary ought to be present in one of the standard
+        # locations during teuthology tests but then testing without
+        # vstart_runner.py will be messy since ceph-fuse binary won't be
+        # present in a standard locations during these sessions. Thus, this
+        # workaround.
+        ceph_fuse_bin_path = 'ceph-fuse' # bin expected to be in env
+        if type(self.mount_a) == 'LocalFuseMount': # for vstart_runner.py runs
+            ceph_fuse_bin_path = join(os_getcwd(), 'bin')
```

Comment:
Shouldn't this be: `ceph_fuse_bin_path = join(os_getcwd(), 'bin', 'ceph-fuse')` ?

Reply:
No. Once the path is exported, ceph-fuse will become available in environment.